genesis/sensors/tactile.py

```diff
@@ -25,83 +28,224 @@ class RigidContactSensor(Sensor):
         The index of the link to which this sensor is attached. If None, defaults to the base link of the entity.
     """
 
-    _all_contacts: Dict[str, Any] | None = None
-    _last_contacts_update_step = -1
+    _cached_link_is_colliding: torch.Tensor | None = None
+    _last_shared_cache_step = -1
 
-    def __init__(self, entity: RigidEntity, link_idx=None):
-        self._cls = self.__class__
-        self._cached = None
+    def __init__(
+        self,
+        entity: RigidEntity,
+        link_idx=None,
+    ):
         self._entity = entity
         self._sim = entity._sim
+        self._solver = self._sim.rigid_solver
         self.link_idx = link_idx if link_idx is not None else entity.base_link_idx
+        self._cached = None
+        self._last_cache_step = -1
 
-    def read(self, envs_idx: Optional[List[int]] = None):
-        if self._cls._last_contacts_update_step == self._sim.cur_step_global:
-            return self._cached
+    def read(self, envs_idx: Optional[List[int]] = None) -> np.ndarray:
+        if type(self)._cached_link_is_colliding is None:
```

Comment:
First, you should never call `type(self)` yourself. Second, relying on class attribute is never a good idea because it is not scoped well enough and many unexpected behaviour may happen. Typically, we support simulating two scenes "in parrallel". Your mechanism will break in such a case. You have no other choice than moving you shared buffer either at scene-level or solver-level.